src/Toolkit/Str.php

```diff
@@ -964,17 +965,18 @@ public static function substr(string $string = null, int $start = 0, int $length
      *
      * </code>
      *
-     * @param string $string The string with placeholders
+     * @param string|null $string The string with placeholders
      * @param array $data Associative array with placeholders as
      *                    keys and replacements as values
-     * @param string $fallback A fallback if a token does not have any matches
+     * @param string|null $fallback A fallback if a token does not have any matches
      * @param string $start Placeholder start characters
      * @param string $end Placeholder end characters
+     * @param \Closure|null $callback A callback to be able to modify each matching result
      * @return string The filled-in string
      */
-    public static function template(string $string = null, array $data = [], string $fallback = null, string $start = '{{', string $end = '}}'): string
+    public static function template(string $string = null, array $data = [], string $fallback = null, string $start = '{{', string $end = '}}', Closure $callback = null): string
```

Comment:
I agree with Nico that the additional argument makes the method signature pretty long and unwieldy.

Idea: What if the current `$fallback` argument supported an `$options` array alternatively? Then the method could be used like this:

```php
Str::template('my {{ template }}', $data, [
    'fallback' => '...',
    'callback' => function() { ... }
]);
```

The argument would still be called `$fallback` for now (renaming it would be a breaking change because of PHP 8's named arguments). Only the `string` type hinting would be removed.

We could then deprecate the previous call signature, add a deprecation warning in 3.6.0 and remove support in 3.7.0.

I think this would be much cleaner in the long run (like `Cookie::set()` where we also introduced `$options` in that way if I remember correctly).

Reply:
I thought and worked on `$fallback` but I gave up because it would be a confusing. The idea of a `$fallback`  that can be a array looks great. I will try.

Reply:
The direct support for the closure is pretty cool for our use case, but I'm not sure if it's good for the future. One argument that can do several things is quite confusing.

What I meant is that `$fallback` gets the double role only temporarily until we have migrated to a `array $options` argument. This would then also support `start` and `end` keys, so in 3.7 the method signature could look like this:

```php
public static function template(?string $string, array $data = [], array $options = []): string
```

Nothing carved in stone though. What do you think @distantnative @bastianallgeier?

Reply:
Yes, it's easier to understand if it's just `string` and `array` for now, and it'll make it even easier to transition in the future. So can we change the name of `$fallback` to `$options` now?

Reply:
I applied it because we can easily revert.  Also, by following the commits and the final version, maybe we can end the discussion faster.